src/main/java/com/google/api/generator/gapic/composer/MethodSampleCodeHelperComposer.java

```diff
@@ -0,0 +1,167 @@
+  }
+
+  // ==================================Helpers===================================================//
+
+  // Assign client variable expr with create client.
+  // e.g EchoClient echoClient = EchoClient.create()
+  private static AssignmentExpr assignClientVariableWithCreateMethodExpr(
+      VariableExpr clientVarExpr) {
+    return AssignmentExpr.builder()
+        .setVariableExpr(clientVarExpr.toBuilder().setIsDecl(true).build())
+        .setValueExpr(
+            MethodInvocationExpr.builder()
+                .setStaticReferenceType(clientVarExpr.variable().type())
+                .setReturnType(clientVarExpr.variable().type())
+                .setMethodName("create")
+                .build())
+        .build();
+  }
+
+  // Create a Map where key is method's argument name, and value is its VariableExpr.
+  private static Map<String, VariableExpr> createMethodArgumentsVariableExprs(
+      List<MethodArgument> arguments) {
+    return arguments.stream()
+        .collect(
+            Collectors.toMap(
+                methodArg -> methodArg.name(),
+                methodArg ->
+                    createVariableExpr(
+                        JavaStyle.toLowerCamelCase(methodArg.name()), methodArg.type())));
+  }
+
+  // Return a list of AssignmentExpr for method argument with its default value.
+  private static List<Expr> assignMethodArgumentsWithDefaultValues(
+      List<MethodArgument> arguments,
+      Map<String, VariableExpr> argVarExprs,
+      Map<String, ResourceName> resourceNames) {
+    return arguments.stream()
+        .map(
+            arg ->
+                createAssignmentExpr(
+                    argVarExprs.get(arg.name()),
+                    DefaultValueComposer.createDefaultValue(arg, resourceNames)))
+        .collect(Collectors.toList());
+  }
+
+  private static Expr createAssignExprForVariableWithClientMethod(
+      VariableExpr variableExpr,
+      VariableExpr clientVarExpr,
+      String methodName,
+      List<Expr> argumentsVarExprs) {
+    MethodInvocationExpr clientMethodInvocationExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(clientVarExpr)
+            .setMethodName(JavaStyle.toLowerCamelCase(methodName))
+            .setArguments(argumentsVarExprs)
+            .setReturnType(variableExpr.variable().type())
+            .build();
+    return AssignmentExpr.builder()
+        .setVariableExpr(variableExpr.toBuilder().setIsDecl(true).build())
+        .setValueExpr(clientMethodInvocationExpr)
+        .build();
+  }
+
+  private static String getClientName(TypeNode clientType) {
+    return JavaStyle.toLowerCamelCase(clientType.reference().name());
+  }
+
+  private static boolean isProtoEmptyType(TypeNode type) {
+    return type.reference().pakkage().equals("com.google.protobuf")
+        && type.reference().name().equals("Empty");
+  }
+
+  private static AssignmentExpr createAssignmentExpr(VariableExpr variableExpr, Expr valueExpr) {
```

Comment:
Same as the comment below.

Reply:
Same as `createVariableExpr`

Reply:
PTAL at the information hiding concerns above.

Reply:
SG